base_layer/wallet/src/storage/sqlite_db/wallet.rs

```diff
@@ -631,24 +592,71 @@ impl WalletBackend for WalletSqliteDatabase {
 /// Master Public Key that is stored in the db
 fn check_db_encryption_status(
     database_connection: &WalletDbConnection,
-    passphrase: Option<SafePassword>,
-) -> Result<Option<XChaCha20Poly1305>, WalletStorageError> {
+    passphrase: SafePassword,
+) -> Result<XChaCha20Poly1305, WalletStorageError> {
     let start = Instant::now();
     let conn = database_connection.get_pooled_connection()?;
     let acquire_lock = start.elapsed();
 
     let db_passphrase_hash = WalletSettingSql::get(&DbKey::PassphraseHash, &conn)?;
     let db_encryption_salt = WalletSettingSql::get(&DbKey::EncryptionSalt, &conn)?;
 
-    let cipher = match (passphrase, db_passphrase_hash, db_encryption_salt) {
-        (Some(_), None, _) => {
-            error!(
-                target: LOG_TARGET,
-                "Passphrase is provided but database is not encrypted"
-            );
-            return Err(WalletStorageError::InvalidPassphrase);
+    let cipher = match (db_passphrase_hash, db_encryption_salt) {
+        (None, None) => {
```

Comment:
We will clean up this code in a posterior PR, which goal is to change the `Backend` api. Doing so here, would require a lot of further refactoring.